### ISSUE TITLE
Rewrite celery.utils.imports in Rust to speed up boot time

### ISSUE THREAD

**Issue by thedrow** [celery/celery#5372]
# Checklist

- [x] I have checked the issues list for similar or identical enhancement to an existing feature.
- [x] I have checked the commit log to find out if a the same enhancement was already implemented in master.

# Brief Summary
The [celery.utils.imports](https://github.com/celery/celery/blob/master/celery/utils/imports.py) module is used when we boot up Celery. Given the fact that we currently have [multiple memory leaks](https://github.com/celery/celery/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+memory+leak) users restart Celery often.
Ensuring Celery's boot time is faster can elevate the situation for the time being.  

# Design

### Risk
Implementing this module in PyO3 isn't going to be easy. It's not currently as documented as it should be.
We need to ensure that this module functions exactly like before and also is faster in loading the configuration and task modules.

## Proposed Behavior
No change in behavior. 

## Proposed UI/UX

No change in UI/UX.

## Diagrams

N/A

## Alternatives

None

**Comment by auvipy** [comment]
currently, I am not in favor of adding another language based code into celery. We should wait for more to move to this direction.

**Comment by thedrow** [comment]
Why not?

**Comment by auvipy** [comment]
THE TECHNOLOGIES ARE NOT MATURE ENOUGH :)

**Comment by thedrow** [comment]
I disagree. By the time we'll implement this, PyO3 will be much more stable.

**Comment by auvipy** [comment]
Is it widely used and adopted? if the answer is no please keep it in your private fork :) celery is a mature platform so it should dog food from the mature ecosystem. rust is also from a very different from python. you can also release a rust version of this.

**Comment by othalla** [comment]
i think the cons of adding a new language to celery it more important than the small performance gains at boot time.

**Comment by auvipy** [comment]
exactly, at least when your contract will end, there will be no one to maintain rust code.

**Comment by thedrow** [comment]
I strongly disagree.
There are multiple reasons to use Rust and they outweigh the disadvantages.
Rust code is safe, C isn't. Rust prevents data races. PyO3 will stabilize enough by the time we reach Celery 5.
